vars/writeTemplateAppsConfigGruntTask.groovy

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/groovy
+
+def call(body) {
+    def taskFile = 'tasks/configs.js'
+
+    if (!fileExists(taskFile)) {
+        writeFile file: taskFile, text: '''\'use strict\';
+
+module.exports = function(grunt) {
+
+  grunt.config.merge({
+    configs: {
+      files: ["global.json", \'global-forms.json\']
+    }
+  });
+
+  var templateHelper = require(\'./lib/templateHelper\').init(grunt);
+  var _ = require(\'underscore\');
+  var url = require(\'url\');
+
+  function updateTemplateAppConfig(filepath, templateAppConfigs) {
+    var globalJson = grunt.file.readJSON(filepath);
+
+    templateHelper.walkObjects(globalJson.show, function (obj) {
+      if (_.has(obj, \'githubUrl\')) {
+        var templateAppConfig = {};
+        var parsedUrl = url.parse(obj.githubUrl, true, true);
+        var tarry = parsedUrl.path.replace(/.git$/, \'\').replace(/^\\//, \'\').split(\'/\');
+        templateAppConfig.githubUrl = obj.githubUrl;
+        templateAppConfig.baseBranch = obj.repoBranch;
+        templateAppConfig.org = tarry[0];
+        templateAppConfig.name = tarry[1];
+        templateAppConfig.jira = "";
+        templateAppConfig.type = "template-apps";
+        templateAppConfig.labels = {};
+        templateAppConfig.labels.rhmap3 = false;
+        templateAppConfig.labels.rhmap4 = false;
+        templateAppConfigs.push(templateAppConfig);
+      }
+    });
+  }
+
+  grunt.registerMultiTask(\'configs\', \'Generate template app config file\', function () {
+    var fileName = grunt.option(\'file-name\') || \'COMPONENTS.json\';
+    var templateAppConfigs = [];
+
+    this.filesSrc.forEach(function (filepath) {
+      updateTemplateAppConfig(filepath, templateAppConfigs);
```

Comment:
Similar to the groovy example in `getTemplateAppComponentConfigs.groovy`, I think it could be better if we didn't pass in `templateAppConfigs` here to be modified within the function. Here's what I think you could have instead (after modifying the `updateTemplateAppConfig` function to just return the result rather than pushing it to `templateAppConfigs`):

```js
const templateAppConfigs = this.filesSrc.map(updateTemplateAppConfig);
```

What do you think?

Disclaimer: I haven't done any javascript in a while, but I think this should work